src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCommercial.tsx

```diff
@@ -193,6 +208,23 @@ export class ArtworkSidebarCommercial extends React.Component<
   }
 }
 
+interface ArtworkSidebarCommercialProps {
```

Comment:
We use this pattern of creating a container for the content that takes mediator in several places lately. Just from the style perspective I'm wondering if it helps us to decide how we should structure those files. I think in the other place we had all the props defined at the beginning of the file and then had a container definition and actual exported class last. I don't have an opinion on what the order should be but we should probably stick with the same order everywhere to make those files more scannable.

Reply:
I'll merge this so we can see which one feels better?

This interface didn't need to be exported, and I put it next to the React component whose props match the interface. When we need an intermediate wrapper like this, it means you may have two sets of components (and two interfaces, state, etc - possibly).

I guess the question is all interfaces at the top and then components, or co-locate the props w/ the component, within a file?

Maybe we should avoid these intermediate components that add context and stuff, except when necessary (since we can do `ContextConsumer` in the main component's `render` method, I think having the intermediate one is more for style)?

Reply:
I think in most cases where we used it - it's because the main component have several rendering functions and each one needed a mediator. So we wrapped this component in a container that takes mediator as a prop and makes it accessible for all the functions instead of passing the mediator around. 

That is my understanding.
So yea - I think there are 2 questions:
- should we be using this pattern at all much?
- and if yes - what is the style recommendation - keep all interfaces together up top and then class declarations below with the wrapper component that is actually rendered being last before graphql definition? Or some other order?

Reply:
I tend to think that the interfaces should be very near to where they're being used -- similar to inlining it: 

```js
export const ArtworkSidebarCommercial<{ foo: string }> { ... }
```